programs/uxd/src/instructions/mint.rs

```diff
@@ -203,24 +204,26 @@ struct DepositoryInfoForMint<'info> {
 
 pub(crate) fn handler(ctx: Context<Mint>, collateral_amount: u64) -> Result<()> {
     // Gather all the onchain states we need (caps, weights and supplies)
-    let controller = ctx.accounts.controller.load()?;
+    let mut controller = ctx.accounts.controller.load_mut()?;
     let identity_depository = ctx.accounts.identity_depository.load()?;
     let mercurial_vault_depository = ctx.accounts.mercurial_vault_depository.load()?;
     let credix_lp_depository = ctx.accounts.credix_lp_depository.load()?;
 
+    // Make sure minting reduces the counter of outflow (minting is inflow)
+    controller.epoch_outflow_amount = controller
+        .epoch_outflow_amount
+        .saturating_sub(collateral_amount);
```

Comment:
um...if there are more inflow than outflow momentarily in this epoch, should this counter goes negative i.e. `epoch_outflow_amount` is of type i64/i128?

Reply:
so actually this is using "saturating_sub" which does not go negative for u64 :) I believe this should just be capped to zero in that case